### PR TITLE
3785 - Value Aggregate: Add: Data.refreshTableDataViews and world data

### PR DESCRIPTION
**Pull request by sorja**
Includes world data

eg.

| country\_iso | variable\_name | col\_name | value |
| :--- | :--- | :--- | :--- |
| WO | forestArea | 2010 | {"raw": "redacted", "faoEstimate": true} |
| WO | forestArea | 2015 | {"raw": "redacted", "faoEstimate": true} |
| WO | forestArea | 2020 | {"raw": "redacted", "faoEstimate": true} |
| WO | forestArea | 2025 | {"raw": "redacted", "faoEstimate": true} |
| WO | otherLand | 1990 | {"raw": "redacted", "faoEstimate": true} |
| WO | otherLand | 2000 | {"raw": "redacted", "faoEstimate": true} |
| WO | otherLand | 2010 | {"raw": "redacted", "faoEstimate": true} |
| WO | otherLand | 2015 | {"raw": "redacted", "faoEstimate": true} |
| WO | otherLand | 2020 | {"raw": "redacted", "faoEstimate": true} |
| WO | otherLand | 2025 | {"raw": "redacted", "faoEstimate": true} |
| WO | otherWoodedLand | 1990 | {"raw": "redacted", "faoEstimate": true} |
| WO | otherWoodedLand | 2000 | {"raw": "redacted", "faoEstimate": true} |
| WO | otherWoodedLand | 2010 | {"raw": "redacted", "faoEstimate": true} |
| WO | otherWoodedLand | 2015 | {"raw": "redacted", "faoEstimate": true} |
| WO | otherWoodedLand | 2020 | {"raw": "redacted", "faoEstimate": true} |
| WO | otherWoodedLand | 2025 | {"raw": "redacted", "faoEstimate": true} |
| WO | totalLandArea | 1990 | {"raw": "redacted", "faoEstimate": true} |
| WO | totalLandArea | 2000 | {"raw": "redacted", "faoEstimate": true} |
| WO | totalLandArea | 2010 | {"raw": "redacted", "faoEstimate": true} |
| WO | totalLandArea | 2015 | {"raw": "redacted", "faoEstimate": true} |
| WO | totalLandArea | 2020 | {"raw": "redacted", "faoEstimate": true} |
| WO | totalLandArea | 2025 | {"raw": "redacted", "faoEstimate": true} |

### CODE REVIEW

File: src/server/repository/assessmentCycle/data/createOrReplaceTableDataView.ts [hunk 1]
```diff
@@ -58,11 +58,20 @@ export const createOrReplaceTableDataView = async (props: Props, client: BasePro
...
+                             group by cr.region_code, cv.variable_name, cv.col_name),
+           global_values as (select 'WO' as country_iso,
+                                    variable_name, col_name,
+                                    jsonb_build_object('raw', text(coalesce(sum((value ->> 'raw')::numeric), 0)), 'faoEstimate', true)
+                                    as value
+                      from country_values
+                      group by variable_name, col_name)
+      select country_iso::varchar(3), variable_name, col_name, value
+      from global_values gv
+      union all
```
Comment (sorja) [inline src/server/repository/assessmentCycle/data/createOrReplaceTableDataView.ts]: added world data

File: src/server/controller/cycleData/data/index.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { refreshTableDataViews } from './refreshTableDataViews'
+
+export const Data = {
```
Comment (minotogna) [inline src/server/controller/cycleData/data/index.ts]: maybe lets call this TableData ?
and the method refreshViews ?
what do u think?

Comment (sorja) [reply src/server/controller/cycleData/data/index.ts]: sure, sounds good

File: src/server/controller/cycleData/tableData/refreshViews.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  assessment: Assessment
+  cycle: Cycle
+}
+
+export const refreshViews = async (props: Props, client: BaseProtocol = DB): Promise<void> => {
+  const { assessment, cycle } = props
+  const tables = await TableRepository.getMany({ assessment, cycle })
+
+  await Promise.all(
+    tables.map(async (table) => DataRepository.createOrReplaceTableDataView({ assessment, cycle, table }, client))
```
Comment (minotogna) [inline src/server/controller/cycleData/tableData/refreshViews.ts]: the generation of the views should not accept any transaction. DDL must be executed independently

**Review by minotogna — APPROVED** [review]

**Pull request merged**